app/models/concerns/mounted_attr.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2023, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas.
+
+module MountedAttr
+  extend ActiveSupport::Concern
+
+  included do
+    class_attribute :mounted_attr_categories
+  end
+
+  module ClassMethods
+    cattr_reader :store
+    @@store = ::MountedAttributes::Store.new
+
+    def mounted_attr(attr, attr_type, options = {})
+      config = store.register(self, attr, attr_type, options)
+
+      define_mounted_entry_getter(config)
+
+      if config.attr_type == :picture
+        define_mounted_picture(config)
+      else
+        define_mounted_attr_getter(config)
+        define_mounted_attr_setter(config)
+      end
+
+      define_mounted_attr_validations(config)
+    end
+
+    private
+
+    def define_mounted_entry_getter(config)
+      define_method("mounted_#{config.attr_name}") do
+        (instance_variable_get("@mounted_#{config.attr_name}") ||
+         instance_variable_set("@mounted_#{config.attr_name}",
+                               MountedAttribute.find_or_initialize_by(
+                                 entry_id: self.id,
+                                 entry_type: config.target_class,
+                                 key: config.attr_name
+                               ))
+        )
+      end
+    end
+
+    def define_mounted_picture(config)
+      class_eval do
+        has_one_attached config.attr_name
```

Comment:
Funktioniert das auch wenn ich z.B. auf mehreren Group Types ein mounted_picture mit demselben Namen definiere?

Reply:
Uhmm.. Also das `has_one_attached` wird auf die Klasse in der es definiert wird gecalled.
Sprich du kannst für Gruppentyp A und Gruppentyp B beides `mounted_attr :logo, :picture` definieren und es ist jeweils was eigenes und sollte problemfrei funktionieren.
Das `ActiveStorage::Attached::One` zeigt immer auf den STI Record und nicht auf die generische `Group` Superklasse oder so